timur/db/migrations/006_add_views.rb

```diff
@@ -0,0 +1,82 @@
+require 'json'
+
+def attribute_type(att)
+  case att[:attribute_class]
+  when nil
+    'magma'
+  when 'MarkdownAttribute'
+    'markdown'
+  else
+    'plot'
+  end
+end
+
+def attribute_options(att)
+  case attribute_type(att)
+  when 'magma', 'markdown'
+    { attribute_name: att[:name] }
+  when 'plot'
+    { plot_id: att[:plot_id] }
+  else
+    {}
+  end
+end
+
+Sequel.migration do
+  up do
+    create_table(:views) do
+      primary_key :id
+      String :model_name, null: false
+      String :project_name, null: false
+      json :document, null: false
+      DateTime :created_at, null: false
+      DateTime :updated_at, null: false
+
+      index [:model_name, :project_name], name: :index_views_on_project_and_model_name, unique: true
+    end
+
+
+    db = Timur.instance.db
+
+    panes = db[:view_panes].all
+    attributes = db[:view_attributes].all
```

Comment:
I guess, the critical question is; do all these view_attributes line up to entries in the model.template in all cases?  Seeing similar issues to @coleshaw